pkg/storage/tsdb/bucketindex/index.go

```diff
@@ -111,6 +114,10 @@ func (m *Block) GetUploadedAt() time.Time {
 // The returned meta doesn't include all original meta.json data but only a subset
 // of it.
 func (m *Block) ThanosMeta() *block.Meta {
+	var compactionHints []string
+	if m.OutOfOrder {
+		compactionHints = []string{tsdb.CompactionHintFromOutOfOrder}
+	}
 	return &block.Meta{
 		BlockMeta: tsdb.BlockMeta{
```

Comment:
`tsdb.BlockMeta` has an `OutOfOrder field that I think should be set as well, no?

Reply:
Talking to @pstibrany, apparently that's legacy and not set anymore, just the hint is used now.

Reply:
Ah, in that case can you change how we're setting it for metric purposes [here](https://github.com/grafana/mimir/blob/35928490d9370255cada7e608b2160870a9849b2/pkg/storegateway/stats.go#L100) as part of this change?